build-system/babel-plugins/babel-plugin-config-transformer/index.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+let hasThirdPartyRegex = false;
+let hasCdnProxyRegex = false;
+
+// Only executes on the ESM build, intended to use with the Google AMP Cache.
+module.exports = function({types: t}) {
+  return {
+    visitor: {
+      VariableDeclarator(path) {
+        const {node} = path;
+        const {name} = node.id;
+        if (name === 'thirdPartyFrameRegex') {
+          hasThirdPartyRegex = true;
+          path.parentPath.remove();
+        } else if (name === 'cdnProxyRegex') {
+          hasCdnProxyRegex = true;
+          path.parentPath.remove();
+        }
+      },
+      LogicalExpression(path) {
+        if (!hasThirdPartyRegex || !hasCdnProxyRegex) {
```

Comment:
I think the intention here was to only run this on the `config.js` file? These variables are never reset, so they'll continue to be used on every file processed afterwards.

A better option would be to provide a file specific override in the babel config: https://babeljs.io/docs/en/options#overrides

Reply:
That's an excellent point. I'll make this change.

Reply:
I wasn't able to get the override option to work, instead went with an approach based on `pre`/`post` in the babel transform.